lib/main_common.pm

```diff
@@ -406,4 +407,97 @@ sub maybe_load_kernel_tests {
     return 1;
 }
 
+sub load_extra_tests() {
+    # Put tests that filled the conditions below
+    # 1) you don't want to run in stagings below here
+    # 2) the application is not rely on desktop environment
+    # 3) running based on preinstalled image
+    return unless get_var('EXTRATEST');
+    # pre-conditions for extra tests ie. the tests are running based on preinstalled image
+    return if get_var("INSTALLONLY") || get_var("DUALBOOT") || get_var("RESCUECD");
+
+    # setup $serialdev permission and so on
+    loadtest "console/consoletest_setup";
+    loadtest "console/hostname";
+    if (any_desktop_is_applicable()) {
+        if (check_var('DISTRI', 'sle')) {
+            # start extra x11 tests from here
+            loadtest 'x11/vnc_two_passwords';
+            loadtest 'x11/user_defined_snapshot';
+        }
+        elsif (check_var('DISTRI', 'opensuse')) {
+            if (!get_var("NOAUTOLOGIN")) {
+                loadtest "x11/multi_users_dm";
+            }
+            if (gnomestep_is_applicable()) {
+                if (check_var('VERSION', '42.2')) {
+                    # 42.2 feature - not even on Tumbleweed
+                    loadtest "x11/gdm_session_switch";
+                }
+                loadtest "x11/seahorse";
+            }
+        }
+    }
+    else {
+        loadtest "console/check_console_font";
+        loadtest "console/zypper_lr";
+        loadtest "console/openvswitch";
+        # dependency of git test
+        loadtest "console/sshd";
+        loadtest "console/zypper_ref";
+        loadtest "console/update_alternatives";
+        # start extra console tests from here
+        if (!get_var("OFW") && !is_jeos) {
+            loadtest "console/aplay";
+        }
+        if (get_var("FILESYSTEM", "btrfs") eq "btrfs") {
+            loadtest "console/btrfs_autocompletion";
+            if (get_var("NUMDISKS", 0) > 1) {
+                loadtest "console/btrfs_qgroups";
+                loadtest "console/btrfs_send_receive";
+                if (check_var('DISTRI', 'sle')) {
+                    loadtest 'console/snapper_cleanup';
```

Comment:
It used to be: `if (sle_version_at_least('12-SP2')) {`. The test would probably fail on pre-12-SP2?